src/Language/PureScript/CodeGen/JS/Common.hs

```diff
@@ -135,7 +135,47 @@ nameIsJsReserved name =
               , "volatile"
               , "while"
               , "with"
-              , "yield" ]
+              , "yield" ] || properNameIsJsReserved name
 
 moduleNameToJs :: ModuleName -> String
-moduleNameToJs (ModuleName pns) = intercalate "_" (runProperName `map` pns)
+moduleNameToJs (ModuleName pns) =
+  let name = intercalate "_" (runProperName `map` pns)
+  in if properNameIsJsReserved name then "$$" ++ name else name
+
+-- |
+-- Checks whether a proper name is reserved in Javascript.
+--
+properNameIsJsReserved :: String -> Bool
+properNameIsJsReserved name =
```

Comment:
"Reserved" already has a meaning in JavaScript. How about `isJsBuiltin`? Also, do we want to catch name conflicts with built-ins such as `parseInt` and `setTimeout`?